README.md
<a href=""><img src="image/banner.jpg" title="db-connect" alt="db-connect"></a>

# dataBaseConnectionGenerator
`stable build v1.1`
### DB-Connect
> Generate Database connection for your project with no effort

> Graphical User Interface for easy usage

[![Build Status](https://img.shields.io/badge/version-v1.1-green?style=flat-square)](https://github.com/yunghog/dataBaseConnectionGenerator)
[![Python](https://img.shields.io/badge/python-3.7-orange?style=flat-square&logo=python)](https://github.com/yunghog/dataBaseConnectionGenerator)
[![Python](https://img.shields.io/badge/pyqt-5.x-yellow?style=flat-square&logo=python)](https://github.com/yunghog/dataBaseConnectionGenerator)

## Table of Contents
- [Requirements](#requirements)
- [Installation](#installation)
- [Features](#features)
- [Usage](#usage)
- [ChangeLog](#changelog)
- [Snapshot](#snapshot)
- [Support](#support)
- [Contributing](#contributing)

## Requirements
> Python 3.x.x

>PyQt 5.x

### Installing PyQt5
- If you are using Anaconda then PyQt is already installed. If you have old version then you may need to update to PyQt5
- If PyQt is not installed then it can be installed using command
``` shell
pip install PyQt5

```

## Installation
- clone this repo into any directory using url : https://github.com/yunghog/w3bsite13locker.git
 ```shell
git clone https://github.com/yunghog/dataBaseConnectionGenerator.git
```
- open the cloned directory
- run `setup.py` using command
```python
python setup.py
```
## Features
- Generate Database Connection file with no effort
- Graphical User Interface for easy usage
- Commandline script for quick operation
- Lightweight python script
- Easy to handle

## Usage
- To use the Commandline version of dbConnect, run `dbGenCli.py` using command
```python
python dbGenCli.py
```
- Then a prompt will ask to input DB variables. Input the variables and hit enter to generate DBConnection
- To use Graphical version of dbConnect, run 'dbGenGui.py' using command
```python
python dbGenGui.py
```
- Fill the fields and hit `generate` button to generate DBConnection
- Click on `preview` button to preview the code

## ChangeLog
```shell
  # dataBaseConnectionGenerator v1.0
  - generate dbConnection using cli script
  - graphical use interface for easy interaction
  - preview button to open the generated script instantly
  - raise exception for invalid input

  # dataBaseConnectionGenerator v1.1
    - added about window
    - dark theme for GUI
    - preview section format
```

## Snapshot
- `setup.py`
<img src="image/snap1.PNG" title="db-connect" alt="db-connect">

- `dbGenGui.py`
<img src="image/gui.PNG" title="db-connect" alt="db-connect" size="70">

- `about.py`
<img src="image/snap2.PNG" title="db-connect" alt="db-connect" size="70">

## Support
- This section will be updated soon

## Contributing
- 🍴 Fork this repo!
- 👯 Clone this repo to your local machine using `https://github.com/yunghog/dataBaseConnectionGenerator.git`
- Pull requests


changeLog.txt
# dataBaseConnectionGenerator v1.0
  - generate dbConnection using cli script
  - graphical use interface for easy interaction
  - preview button to open the generated script instantly
  - raise exception for invalid input

# dataBaseConnectionGenerator v1.1
  - added about window
  - dark theme for GUI
  - preview section format


about.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'about.ui'
#
# Created by: PyQt5 UI code generator 5.9.2
#
# WARNING! All changes made in this file will be lost!
import webbrowser
from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_Dialog(object):
    def setupUi(self, Dialog):
        Dialog.setObjectName("Dialog")
        Dialog.resize(401, 214)
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(85, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 239, 239))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(63, 63, 63))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Light, brush)
        brush = QtGui.QBrush(QtGui.QColor(47, 47, 47))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Midlight, brush)
        brush = QtGui.QBrush(QtGui.QColor(54, 54, 54))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(57, 57, 57))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.AlternateBase, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 239, 239))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(63, 63, 63))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Light, brush)
        brush = QtGui.QBrush(QtGui.QColor(47, 47, 47))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Midlight, brush)
        brush = QtGui.QBrush(QtGui.QColor(54, 54, 54))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(57, 57, 57))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.AlternateBase, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 239, 239))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(63, 63, 63))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Light, brush)
        brush = QtGui.QBrush(QtGui.QColor(47, 47, 47))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Midlight, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(57, 57, 57))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.AlternateBase, brush)
        Dialog.setPalette(palette)
        Dialog.setAcceptDrops(False)
        icon = QtGui.QIcon.fromTheme("about")
        Dialog.setWindowIcon(icon)
        self.label = QtWidgets.QLabel(Dialog)
        self.label.setGeometry(QtCore.QRect(-10, 0, 411, 101))
        self.label.setText("")
        self.label.setPixmap(QtGui.QPixmap("D:/Photos/Edits/dbConnect.jpg"))
        self.label.setScaledContents(True)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(Dialog)
        self.label_2.setGeometry(QtCore.QRect(110, 110, 191, 21))
        font = QtGui.QFont()
        font.setFamily("Monument Extended")
        font.setPointSize(10)
        self.label_2.setFont(font)
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setObjectName("label_2")
        self.horizontalLayoutWidget = QtWidgets.QWidget(Dialog)
        self.horizontalLayoutWidget.setGeometry(QtCore.QRect(110, 170, 191, 41))
        self.horizontalLayoutWidget.setObjectName("horizontalLayoutWidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget)
        self.horizontalLayout.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.github = QtWidgets.QPushButton(self.horizontalLayoutWidget)
        self.github.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.github.setObjectName("github")
        self.github.clicked.connect(self.opengithub)
        self.horizontalLayout.addWidget(self.github)
        self.insta = QtWidgets.QPushButton(self.horizontalLayoutWidget)
        self.insta.setEnabled(True)
        self.insta.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.insta.setMouseTracking(False)
        self.insta.setObjectName("insta")
        self.insta.clicked.connect(self.openinsta)
        self.horizontalLayout.addWidget(self.insta)
        self.label_3 = QtWidgets.QLabel(Dialog)
        self.label_3.setGeometry(QtCore.QRect(200, 126, 121, 20))
        font = QtGui.QFont()
        font.setFamily("Futura Md BT")
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")

        self.retranslateUi(Dialog)
        QtCore.QMetaObject.connectSlotsByName(Dialog)

    def retranslateUi(self, Dialog):
        _translate = QtCore.QCoreApplication.translate
        Dialog.setWindowTitle(_translate("Dialog", "About"))
        self.label_2.setText(_translate("Dialog", "Author : Samartha"))
        self.github.setText(_translate("Dialog", "Github"))
        self.insta.setText(_translate("Dialog", "Instagram"))
        self.label_3.setText(_translate("Dialog", "Developer & Designer"))
    def opengithub(self):
        webbrowser.open('https://github.com/yunghog')
    def openinsta(self):
        webbrowser.open('https://www.instagram.com/samartha__')

if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    about = QtWidgets.QDialog()
    ui = Ui_Dialog()
    ui.setupUi(about)
    about.show()
    sys.exit(app.exec_())


dbGen.py
#dbGen.py : Easy way to generate Database connection file for your project
#author : Samartha

def gen(format,dbServer,dbUser,dbPswd,dbName):
    flag=0
    if format=='js':
        dbCon = open(r'dbConnect.js','w')
        dbCon.write("var mysql = require('mysql')\n")
        dbCon.write("var con = mysql.createConnection({\n")
        host="host: \"" + dbServer + "\",\n"
        dbCon.write(host)
        dbUser="user: \"" + dbUser + "\",\n"
        dbCon.write(dbUser)
        dbPswd="password: \"" + dbPswd + "\",\n"
        dbCon.write(dbPswd)
        dbCon.write("})")
        dbCon.write("con.connect(function(err)){\n")
        dbCon.write("if(err) throw err;\n")
        dbCon.write("console.log('Connected')\n")
        dbCon.write("});")
        flag=1
    elif format=='php':
        dbCon = open(r'dbConnect.php','w')
        dbCon.write("<?php\n")
        host="$servername = \"" + dbServer + "\";\n"
        dbCon.write(host)
        dbUser="$username = \"" + dbUser + "\";\n"
        dbCon.write(dbUser)
        dbName="$dbname = \"" + dbName + "\";\n"
        dbCon.write(dbName)
        dbPswd="$password = \"" + dbPswd + "\";\n"
        dbCon.write(dbPswd)
        dbCon.write("$conn = new mysqli($servername,$username,$password,$dbname);\n")
        dbCon.write("if ($conn->connect_error) {\n")
        dbCon.write(" die('Connection failed: ' . $conn->connect_error);\n")
        dbCon.write("}\n")
        dbCon.write("echo 'Connected Successfully';\n?>")
        flag=1
    if flag==1:
        return 'Success'
    else:
        return 'Failed'
def uiPrev(format):
    try:
        if format=='php':
            conFile=open('dbConnect.php','r')
        elif format=='js':
            conFile=open('dbConnect.js','r')
        return conFile.read()
    except FileNotFoundError as e:
        return str(e)

def validate(format,dbServer,dbUser,dbPswd,dbName):
    if dbServer=='' or dbUser=='' or dbPswd=='' or dbName=='' :
        return False
    else:
        return True


dbGenGui.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'dbGen.ui'
#
# Created by: PyQt5 UI code generator 5.9.2
#
# WARNING! All changes made in this file will be lost!
import dbGen as d
from about import Ui_Dialog
from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_DBConnectionGen(object):
    def openAbout(self):
        self.aboutWindow = QtWidgets.QDialog()
        self.ui = Ui_Dialog()
        self.ui.setupUi(self.aboutWindow)
        self.aboutWindow.show()
    def setupUi(self, DBConnectionGen):
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(85, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 239, 239))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(63, 63, 63))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Light, brush)
        brush = QtGui.QBrush(QtGui.QColor(47, 47, 47))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Midlight, brush)
        brush = QtGui.QBrush(QtGui.QColor(54, 54, 54))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(57, 57, 57))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.AlternateBase, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 255, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 239, 239))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(63, 63, 63))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Light, brush)
        brush = QtGui.QBrush(QtGui.QColor(47, 47, 47))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Midlight, brush)
        brush = QtGui.QBrush(QtGui.QColor(54, 54, 54))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(57, 57, 57))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.AlternateBase, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 239, 239))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(63, 63, 63))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Light, brush)
        brush = QtGui.QBrush(QtGui.QColor(47, 47, 47))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Midlight, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Base, brush)
        brush = QtGui.QBrush(QtGui.QColor(45, 45, 45))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(57, 57, 57))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.AlternateBase, brush)
        DBConnectionGen.setObjectName("DBConnectionGen")
        DBConnectionGen.resize(497, 600)
        self.centralwidget = QtWidgets.QWidget(DBConnectionGen)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setEnabled(True)
        self.label.setGeometry(QtCore.QRect(10, 10, 471, 31))
        font = QtGui.QFont()
        font.setFamily("Aero")
        font.setPointSize(22)
        font.setItalic(True)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.preview_text = QtWidgets.QTextBrowser(self.centralwidget)
        self.preview_text.setGeometry(QtCore.QRect(15, 350, 471, 221))
        font = QtGui.QFont()
        font.setPointSize(8)
        self.preview_text.setFont(font)
        self.preview_text.setObjectName("preview_text")
        self.layoutWidget = QtWidgets.QWidget(self.centralwidget)
        self.layoutWidget.setGeometry(QtCore.QRect(90, 69, 321, 251))
        self.layoutWidget.setObjectName("layoutWidget")
        self.gridLayout = QtWidgets.QGridLayout(self.layoutWidget)
        self.gridLayout.setContentsMargins(0, 0, 0, 0)
        self.gridLayout.setHorizontalSpacing(32)
        self.gridLayout.setVerticalSpacing(10)
        self.gridLayout.setObjectName("gridLayout")
        self.extention = QtWidgets.QComboBox(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(16)
        self.extention.setFont(font)
        self.extention.setObjectName("extention")
        self.extention.addItem("")
        self.extention.addItem("")
        self.gridLayout.addWidget(self.extention, 0, 1, 1, 1)
        self.server = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(16)
        self.server.setFont(font)
        self.server.setObjectName("server")
        self.gridLayout.addWidget(self.server, 1, 1, 1, 1)
        self.label_3 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setFamily("Bebas")
        font.setPointSize(16)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.gridLayout.addWidget(self.label_3, 2, 0, 1, 1)
        self.password = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(16)
        self.password.setFont(font)
        self.password.setObjectName("password")
        self.gridLayout.addWidget(self.password, 3, 1, 1, 1)
        self.database = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(16)
        self.database.setFont(font)
        self.database.setObjectName("database")
        self.gridLayout.addWidget(self.database, 4, 1, 1, 1)
        self.label_6 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setFamily("Bebas")
        font.setPointSize(16)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.gridLayout.addWidget(self.label_6, 4, 0, 1, 1)
        self.label_2 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setFamily("Bebas")
        font.setPointSize(16)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.gridLayout.addWidget(self.label_2, 1, 0, 1, 1)
        self.generate = QtWidgets.QPushButton(self.layoutWidget)
        self.generate.setFlat(False)
        self.generate.setObjectName("generate")
        self.gridLayout.addWidget(self.generate, 5, 0, 1, 1)
        self.generate.clicked.connect(self.dbGenUi)
        self.label_5 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setFamily("Bebas")
        font.setPointSize(16)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.gridLayout.addWidget(self.label_5, 0, 0, 1, 1)
        self.username = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(16)
        self.username.setFont(font)
        self.username.setObjectName("username")
        self.gridLayout.addWidget(self.username, 2, 1, 1, 1)
        self.preview = QtWidgets.QPushButton(self.layoutWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.preview.sizePolicy().hasHeightForWidth())
        self.preview.setSizePolicy(sizePolicy)
        self.preview.setObjectName("preview")
        self.gridLayout.addWidget(self.preview, 5, 1, 1, 1)
        self.preview.clicked.connect(self.dbConPreview)
        self.label_4 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setFamily("Bebas")
        font.setPointSize(16)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.gridLayout.addWidget(self.label_4, 3, 0, 1, 1)
        DBConnectionGen.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(DBConnectionGen)
        self.statusbar.setAutoFillBackground(False)
        self.statusbar.setSizeGripEnabled(False)
        self.statusbar.setObjectName("statusbar")
        DBConnectionGen.setStatusBar(self.statusbar)
        DBConnectionGen.setPalette(palette)


        self.retranslateUi(DBConnectionGen)
        QtCore.QMetaObject.connectSlotsByName(DBConnectionGen)

    def retranslateUi(self, DBConnectionGen):
        _translate = QtCore.QCoreApplication.translate
        DBConnectionGen.setWindowTitle(_translate("DBConnectionGen", "MainWindow"))
        self.label.setText(_translate("DBConnectionGen", "Database Connection Generator"))
        self.preview_text.setHtml(_translate("DBConnectionGen", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'MS Shell Dlg 2\'; font-size:8pt; font-weight:400; font-style:normal;\">\n"
"<p style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p></body></html>"))
        self.extention.setItemText(0, _translate("DBConnectionGen", "js"))
        self.extention.setItemText(1, _translate("DBConnectionGen", "php"))
        self.label_3.setText(_translate("DBConnectionGen", "Username"))
        self.label_6.setText(_translate("DBConnectionGen", "Database"))
        self.label_2.setText(_translate("DBConnectionGen", "Server"))
        self.generate.setText(_translate("DBConnectionGen", "Generate"))
        self.label_5.setText(_translate("DBConnectionGen", "Extention"))
        self.preview.setText(_translate("DBConnectionGen", "Preview"))
        self.label_4.setText(_translate("DBConnectionGen", "Password"))


    def dbConPreview(self):
        self.openAbout()
        format=self.extention.currentText()
        self.preview_text.setText(d.uiPrev(format))
    def dbGenUi(self):
        format = self.extention.currentText()
        dbServer = self.server.text()
        dbName = self.database.text()
        dbUser = self.username.text()
        dbPswd = self.password.text()
        if d.validate(format,dbServer,dbName,dbPswd,dbName) == True:
            self.preview_text.setText(d.gen(format,dbServer,dbName,dbPswd,dbName))
        else:
            msg = QtWidgets.QMessageBox()
            msg.setWindowTitle("Error!")
            msg.setText("Fields cannot be empty")
            x = msg.exec_()  # this will show our messagebox

if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    DBConnectionGen = QtWidgets.QMainWindow()
    ui = Ui_DBConnectionGen()
    ui.setupUi(DBConnectionGen)
    DBConnectionGen.show()
    sys.exit(app.exec_())
